plugins/uefi-capsule/fu-uefi-capsule-plugin.c

```diff
@@ -569,19 +561,26 @@ fu_uefi_capsule_plugin_is_esp_linux(FuVolume *esp, GError **error)
 		const gchar *fn = g_ptr_array_index(files, i);
 		g_autofree gchar *basename = g_path_get_basename(fn);
 		g_autofree gchar *basename_lower = g_utf8_strdown(basename, -1);
-		if (g_strv_contains((const gchar *const *)basenames, basename_lower)) {
+
+		for (guint j = 0; prefixes_root[j] != NULL; j++) {
+			if (!g_str_has_prefix(basename_lower, prefixes_root[j]))
+				continue;
+			if (!g_str_has_suffix(basename_lower, ".efi"))
+				continue;
 			g_info("found %s which indicates a Linux ESP, using %s", fn, mount_point);
 			return TRUE;
 		}
 	}
 
 	/* failed */
-	basenames_str = g_strjoinv("|", (gchar **)basenames);
+	for (guint i = 0; prefixes_root[i] != NULL; i++)
+		prefixes[i] = g_strdup_printf("%s*.efi", prefixes_root[i]);
+	prefixes_str = g_strjoinv("|", (gchar **)prefixes);
```

Comment:
i don't think adding the .efi here is useful; we're looking for the prefixes, not the files -- and it think it would be confusing to show a list of things we're looking for when we're really being a lot more permissive. Maybe `g_strdup_printf("%s*"` might be clearer.